src/Features/PhotosPage.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {createRef, DetailedHTMLProps, InputHTMLAttributes} from 'react';
+import {API} from "../API/Api";
+import {setLoadStatusAC, setLoadingStatusAC} from '../Store/LoadStatusReducer'
+import {useDispatch, useSelector} from "react-redux";
+import {AppStoreType} from "../Store/State";
+import {stateItem} from "../Store/ImagesReducer"
+import {InitialStateType} from "../Store/LoadStatusReducer"
+import {addDataAC} from "../Store/ImagesReducer";
+import "./marsGallery.css"
+import Heart from "./Heart";
+import {addFavoriteAC} from "../Store/FavoriteReducer";
+
+export type ImgPropsType={id: number, img_src: string}
+
+type DefaultInputPropsType = DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
+type InputRefType = DefaultInputPropsType & { current: any };
+
+
+function PhotosPage() {
+
+    const inputRef: InputRefType = createRef();
+
+    const Images=useSelector<AppStoreType, stateItem[]>(state => state.Images)
+    const Status=useSelector<AppStoreType, InitialStateType>(state => state.Status)
+    const dispatch=useDispatch()
+
+
+    const clickHandler = () => {
+        dispatch(setLoadingStatusAC())
+        API.getPhotos(inputRef.current.value)
```

Comment:
this code should be inside action creator with usual thunk usage or createAsyncThunk